modules/post/windows/gather/checkvm.rb

```diff
@@ -170,24 +183,56 @@ def xen?
       return true if service_exists?(service)
     end
 
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemProductName') =~ /xen/i
+
     false
   end
 
   def qemu?
-    key_path = 'HKLM\HARDWARE\DEVICEMAP\Scsi\Scsi Port 0\Scsi Bus 0\Target Id 0\Logical Unit Id 0'
+    key_path = 'HKLM\\HARDWARE\\DEVICEMAP\\Scsi\\Scsi Port 0\\Scsi Bus 0\\Target Id 0\\Logical Unit Id 0'
     return true if registry_getvaldata(key_path, 'Identifier') =~ /qemu|virtio/i
 
-    key_path = 'HKLM\HARDWARE\DESCRIPTION\System\CentralProcessor\0'
+    key_path = 'HKLM\\HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0'
     return true if registry_getvaldata(key_path, 'ProcessorNameString') =~ /qemu/i
 
-    %w[HKLM\HARDWARE\ACPI\DSDT HKLM\HARDWARE\ACPI\FADT HKLM\HARDWARE\ACPI\RSDT].each do |key|
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'VideoBiosVersion') =~ /qemu/i
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System\\BIOS', 'SystemManufacturer') =~ /qemu/i
+
+    %w[HKLM\\HARDWARE\\ACPI\\DSDT HKLM\\HARDWARE\\ACPI\\FADT HKLM\\HARDWARE\\ACPI\\RSDT].each do |key|
       srvvals = registry_enumkeys(key)
       return true if srvvals && srvvals.include?('BOCHS_')
     end
 
     false
   end
 
+   def parallels?
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'SystemBiosVersion') =~ /parallels/i
+    return true if registry_getvaldata('HKLM\\HARDWARE\\DESCRIPTION\\System', 'VideoBiosVersion') =~ /parallels/i
+
+    false
+  end
+
+  def joesandbox?
+    vpcprocs = [
+      'joeboxcontrol.exe',
+      'joeboxserver.exe'
+    ]
+    get_processes.each do |x|
+      vpcprocs.each do |p|
+        return true if p == x['name'].downcase
+      end
+    end
+
+    key_path = 'HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion'
+    return true if registry_getvaldata(key_path, 'ProductId') == '55274-640-2673064-23950'
+    key_path = 'HKLM\\SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion'
+    return true if registry_getvaldata(key_path, 'ProductId') == '55274-640-2673064-23950'
```

Comment:
Are these product IDs otherwise invalid or is there a chance there will be a false positive if someone lands in this particular version?

A comment noting what those `ProductId`s are would be useful as well.

Reply:
https://duckduckgo.com/?q=%2255274-640-2673064-23950%22&ia=web is "documented" as JoeSandbox